### PR TITLE
Add more tests for fixed ICEs

### LINKED ISSUE #36122: ICE when accessing externed DST
I realize it's kinda nonsensical, but here:

``` rust
fn main() {
    extern {
        static symbol: [usize];
    }
    println!("{}", symbol[0]);
}
```

```
thread 'rustc' panicked at 'assertion failed: self.llextra != ptr::null_mut()', ../src/librustc_trans/mir/lvalue.rs:64
stack backtrace:
   1:     0x7f8aedf3eb13 - std::sys::backtrace::tracing::imp::write::h46f28e67d38b4637
   2:     0x7f8aedf4f40d - std::panicking::default_hook::{{closure}}::h1d3243f546573ff4
   3:     0x7f8aedf4d95d - std::panicking::default_hook::h96c288d728df3ebf
   4:     0x7f8aedf4e058 - std::panicking::rust_panic_with_hook::hb1322e5f2588b4db
   5:     0x7f8aed023f9f - std::panicking::begin_panic::h0914615a412ba184
   6:     0x7f8aed0fd12b - rustc_trans::mir::lvalue::LvalueRef::len::h7704b2160ae8eef8
   7:     0x7f8aed10216f - rustc_trans::mir::rvalue::<impl rustc_trans::mir::MirContext<'bcx, 'tcx>>::trans_rvalue_operand::hc6bf66aa2cc81d96
   8:     0x7f8aed0ed8e4 - rustc_trans::mir::block::<impl rustc_trans::mir::MirContext<'bcx, 'tcx>>::trans_block::he2f6252e2877e76b
   9:     0x7f8aed0eb438 - rustc_trans::mir::trans_mir::hc6963a43e5e7752b
  10:     0x
7f8aed08aeae - rustc_trans::base::trans_closure::h2174ffc103f3d309
  11:     0x7f8aed10de82 - rustc_trans::trans_item::TransItem::define::h159a1f0a8708f3d3
  12:     0x7f8aed08e3b3 - rustc_trans::base::trans_crate::hf067cf9d0a8bcbf2
  13:     0x7f8aee308962 - rustc_driver::driver::phase_4_translate_to_llvm::he9d5d0022988d46e
  14:     0x7f8aee3439a0 - rustc_driver::driver::compile_input::{{closure}}::h13e3af2358852a64
  15:     0x7f8aee340450 - rustc_driver::driver::phase_3_run_analysis_passes::{{closure}}::h63e83c14f91c1ee5
  16:     0x7f8aee286a5f - rustc::ty::context::TyCtxt::create_and_enter::h6d5dc8f8bc2765a0
  17:     0x7f8aee2f7f64 - rustc_driver::driver::compile_input::h7dacd98cd2fd7d2b
  18:     0x7f8aee31e05d - rustc_driver::run_compiler::h37c4294ab73436f7
  19:     0x7f8aee25fbb3 - std::panicking::try::do_call::h4d040997e2efdaf3
  20:     0x7f8aedf5d626 - __rust_maybe_catch_panic
  21:     0x7f8aee27cf19 - <F as alloc::boxed::FnBox<A>>::call_box::hefd1c85fab3e6c31
  22:     0x7f8aedf4be90 - std::sys::thread::Thread::new::thread_start::h5b631f48cd23f128
  23:     0x7f8ae63b06f9 - start_thread
  24:     0x7f8aedb8fb5c - clone
  25:                0x0 - <unknown>
```

```
rustc 1.13.0-nightly (acd3f796d 2016-08-28)
binary: rustc
commit-hash: acd3f796d26e9295db1eba1ef16e0d4cc3b96dd5
commit-date: 2016-08-28
host: x86_64-unknown-linux-gnu
release: 1.13.0-nightly
```

### LINKED ISSUE #58094: assertion failed: bpos.to_u32() >= mbc.pos.to_u32() + mbc.bytes as u32'
Reproduction branch: https://github.com/bjorn3/rustc_codegen_cranelift/tree/syntax_assertion_fail

When applying the change from the last commit I get a panic in my code during the call to `SourceMap::lookup_char_pos`.

```
$ rustc -V
rustc 1.34.0-nightly (8a57831a4 2019-02-02)
$ uname -a
Darwin iMac.local 18.2.0 Darwin Kernel Version 18.2.0: Mon Nov 12 20:24:46 PST 2018; root:xnu-4903.231.4~2/RELEASE_X86_64 x86_64
```

<details>
<summary>Backtrace</summary>

```
thread 'rustc' panicked at 'assertion failed: bpos.to_u32() >= mbc.pos.to_u32() + mbc.bytes as u32', src/libsyntax/source_map.rs:832:17
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
   1: std::sys_common::backtrace::_print
   2: std::panicking::default_hook::{{closure}}
   3: std::panicking::default_hook
   4: rustc::util::common::panic_hook
   5: std::panicking::rust_panic_with_hook
   6: std::panicking::begin_panic
   7: syntax::source_map::SourceMap::bytepos_to_file_charpos
   8: syntax::source_map::SourceMap::lookup_char_pos
   9: rustc_codegen_cranelift::debuginfo::FunctionDebugContext::define::{{closure}}
  10: rustc_codegen_cranelift::debuginfo::FunctionDebugContext::define
  11: rustc_codegen_cranelift::base::trans_fn::{{closure}}::{{closure}}
  12: <core::option::Option<T>>::map
  13: rustc_codegen_cranelift::base::trans_fn::{{closure}}
  14: <cranelift_module::module::Module<B>>::define_function_peek_compiled
  15: rustc_codegen_cranelift::base::trans_fn
  16: rustc_codegen_cranelift::base::trans_mono_item
  17: rustc_codegen_cranelift::codegen_mono_items::{{closure}}::{{closure}}
  18: rustc_codegen_cranelift::unimpl::try_unimpl::{{closure}}
  19: <std::panic::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
  20: std::panicking::try::do_call
  21: __rust_maybe_catch_panic
  22: std::panicking::try
  23: std::panic::catch_unwind
  24: rustc_codegen_cranelift::unimpl::try_unimpl
  25: rustc_codegen_cranelift::codegen_mono_items::{{closure}}
  26: rustc_codegen_cranelift::time
  27: rustc_codegen_cranelift::codegen_mono_items
  28: rustc_codegen_cranelift::codegen_cgus
  29: <rustc_codegen_cranelift::CraneliftCodegenBackend as rustc_codegen_utils::codegen_backend::CodegenBackend>::codegen_crate
  30: rustc::util::common::time
  31: rustc_driver::driver::phase_4_codegen
  32: rustc_driver::driver::compile_input::{{closure}}
  33: <std::thread::local::LocalKey<T>>::with
  34: rustc::ty::context::TyCtxt::create_and_enter
  35: rustc_driver::driver::compile_input
  36: rustc_driver::run_compiler_with_pool
  37: <scoped_tls::ScopedKey<T>>::set
  38: rustc_driver::run_compiler
  39: <scoped_tls::ScopedKey<T>>::set
```

</details>

### PR DESCRIPTION
**Pull request by JohnTitor**
Closes #36122 (fixed in 1.20.0)
Closes #58094 (fixed in #66054)
Also, fix mistaken test case, from #30904 to #30906 (cc @eddyb)

r? @Centril

### CODE REVIEW

File: src/test/ui/parser/missing_right_square_bracket.rs [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// ignore-tidy-trailing-newlines
```
Comment (Centril) [inline src/test/ui/parser/missing_right_square_bracket.rs]: Can you please mention the PR as a comment and the issue number in the file name?

**Comment by Centril** [comment]
r=me rollup with comment ^-- addressed

File: src/test/ui/parser/missing_right_square_bracket.rs (hunk 1 above)
Comment (JohnTitor) [reply src/test/ui/parser/missing_right_square_bracket.rs]: Added comment and renamed file name.

File: src/test/ui/parser/issue-58094-missing-right-square-bracket.rs [hunk 2]
```diff
@@ -0,0 +1,4 @@
+// https://github.com/rust-lang/rust/issues/58094
```
Comment (Centril) [inline src/test/ui/parser/issue-58094-missing-right-square-bracket.rs]:
```suggestion
// Fixed in #66054.
```

**Comment by JohnTitor** [comment]
Ugh, applying suggestion inserted a new line for some reason.

**Comment by Centril** [comment]
:+1: @bors r+

**Comment by JohnTitor** [comment]
Failed in https://github.com/rust-lang/rust/pull/66432#issuecomment-554259378
@bors r-

**Comment by Centril** [comment]
You'll need to do `--bless --compare-mode=nll` to fix that error. r=me when done.

**Comment by Centril** [comment]
Let's give it another go, @bors r+

**Pull request merged**